### ISSUE TITLE
CreateService error: host port mapping for the container port is required

### ISSUE THREAD

**Issue by yusinto** [hashicorp/terraform#8490]
### Terraform Version
0.7.1

### Affected Resource(s)
- aws_ecs_service
- aws_ecs_task_definition

### Terraform Configuration Files
```hcl
resource "aws_ecs_cluster" "WebsiteCluster" {
  name = "${var.tag_name}"
}

resource "aws_ecs_task_definition" "WebsiteTaskDefinition" {
  family = "${var.tag_name}"

  container_definitions = <<EOF
[
    {
        "image": "domain/repository/website:latest",
        "name": "${var.tag_name}",
        "cpu": 0,
        "memory": 512,
        "essential": true,
        "portMappings": [
            {
                "hostPort": 0, // It seems this is the source of the issue
                "containerPort": 80,
                "protocol": "tcp"
            }
        ],
        "command": [
            "test"
        ],
        "logConfiguration": {
            "logDriver": "awslogs",
            "options": {
                "awslogs-group": "${var.tag_name}",
                "awslogs-region": "ap-southeast-2"
            }
        }
    }
]
EOF
}

// ERROR when creating service: aws_ecs_service.WebsiteEcsService: InvalidParameterException: A host port mapping for the container port 80 is required
resource "aws_ecs_service" "WebsiteEcsService" {
  name = "${var.tag_name}"
  cluster = "${aws_ecs_cluster.WebsiteCluster.id}"
  task_definition = "${aws_ecs_task_definition.WebsiteTaskDefinition.arn}"
  desired_count = 2
  iam_role = "ecsServiceRole"
  depends_on = ["aws_alb_listener.WebsiteAlbListener"]

  load_balancer {
    elb_name = "${aws_alb.WebsiteAlb.name}"
    container_name = "${var.tag_name}"
    container_port = 80
  }
}
```

### Debug Output
https://gist.github.com/yusinto/c9dad1c91098bd030b011f103357d8e4

### Expected Behavior
Ecs service should have been created with the specified task definition with an alb load balancing across those containers using dynamic ports

### Actual Behavior
I am trying to use the ALB dynamic host port mapping with my containers by setting hostPort: 0 in my task definition file. However I get the error below when I tried to create the ECS service using that task definition:
aws_ecs_service.WebsiteEcsService: InvalidParameterException: A host port mapping for the container port 80 is required

### Steps to Reproduce
1. `terraform apply`

### Important Factoids
Alb is a recent addition to aws and hence terraform, so perhaps this is a bug that was undetected because not many people have adopted alb.

**Comment by yusinto** [comment]
Fixed in 0.7.2. The problem was alb was not supported in ecs in 0.7.1. Sorry for the trouble. Thanks a lot for your efforts in this awesome project @stack72 @catsby

**Comment by catsby** [comment]
Thanks for following up! I apologize that you didn't get any kind of reply until later 😦
